server/src/main/resources/openapi/reservoir.yaml

```diff
@@ -506,6 +506,56 @@ paths:
           $ref: "#/components/responses/trait_404"
         "500":
           $ref: "#/components/responses/trait_500"
+  /reservoir/upload:
+    parameters:
+      - $ref: headers/okapi-tenant.yaml
+      - $ref: headers/okapi-token.yaml
+      - $ref: headers/okapi-url.yaml
+      - in: query
+        name: sourceId
+        required: true
+        description: source identifier (library ID)
+        schema:
+          type: string
+      - in: query
+        name: sourceVersion
+        description: source version
+        schema:
+          type: integer
+          default: 1
+      - in: query
+        name: localIdPath
+        description: JSON-path for where local identifier is fetched from
+        schema:
+          type: string
+    post:
+      description: upload files.
+      operationId: uploadFiles
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                records:
```

Comment:
I propose we change this field name to `files` or `upload`, `records` is confusing because these are not "records" in the Reservoir sense.

Reply:
In Reservoir "records" or "global records" is something very specific and described by the following schema: https://github.com/folio-org/mod-reservoir/blob/master/server/src/main/resources/openapi/schemas/globalRecord.json
 
Files that are uploaded through this form do not contain entries of this type and there's a conversion process needed to transform the contents into what we refer to as "records". For MFHD after conversion, the "totals" will not even match. 

I am also thinking ahead: once we add persistence for files and a "File" management API (to retrieve and delete uploaded files) it seems natural that we use terminology that refers to the whole units we manage, and not their contents.

Reply:
The purpose of the service is to generate global records. Not upload files. The arguments sourceId, sourceVersion, localId, records are the arguments.

If we want an upload files facility, then just need files. And for the record. I don't think mod-reservoir is a file storage. Another module can do that. *that* module can then feed records from files via what we have now or other. This is also what David is doing by hand now.

Reply:
I disagree, as this is a file upload facility for all practical purposes, especially from the client/browser standpoint.

I agree that we can consider file upload to be a separate service but then it will communicate with mod-reservoir via the regular "record ingest API" and will not need this as this API is only useful for clients like "curl" or the "browser".  If we provide a seperate service then we should move this entire API along wiht the arguments.

I don't want to hold this back so I will unblock the PR but I think the naming is incosistent and confusing (e.g the operation in the YAML is called "uploadFiles" while in the code it's called "uploadRecords).

Reply:
Files is a generic term. Admitted, records, are too. If we had a facility to upload JavaScript via a form would that also be called files? The file is the transport that is used by curl or a browser. As soon as it leaves there it's network.